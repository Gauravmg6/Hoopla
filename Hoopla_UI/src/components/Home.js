import React, { Component } from 'react';
import Carousel from 'react-material-ui-carousel';
import { Link,BrowserRouter } from 'react-router-dom';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import axios from "axios";
import Cardview from './Cardview';
const url = "http://localhost:3000/";

class Home extends Component {
  constructor(props){
    super(props);
  this.state={
    category:"",
    cardData:"",
    errorMessage:""
  }
}
  getData = (data) => {
    this.setState({cardData:""});
    axios.get(url+data)
      .then(response => {this.setState({cardData: response.data})
      console.log(response.data)}
      )
      .catch(error => {
        this.setState({ errorMessage: error.response ? error.response.data.message : error.message })
      })
  }
  getCategory=(e)=>{
    e.preventDefault();
    var data=e.target.getAttribute("value");
    this.getData(data);
  }


  viewDetails=()=>{
    var object=JSON.parse(sessionStorage.getItem("data"));
    sessionStorage.removeItem("data")
    if(this.state.cardData!=""){
      return <Cardview obj={this.state.cardData}></Cardview>
    }
    else if(object!=null){
      return <Cardview obj={object}></Cardview>
    }
  }
  render() {
    if(this.state.cardData==="" && sessionStorage.getItem("data")==null){
     return (
      <React.Fragment>
        {/* Slider */}
        <AppBar position="static">
          <Toolbar>
            
            <Link name="category" className="text-white pl-5" style={{fontSize:22,textDecoration:"none"}} value="Electronics" onClick={this.getCategory}>ELECTRONICS</Link>
            <Link name="category" className="ml-auto text-white" style={{fontSize:22,textDecoration:"none"}} value="Shoes" onClick={this.getCategory} >SHOES</Link>
            <Link name="category" className="ml-auto text-white" style={{fontSize:22,textDecoration:"none"}} value="Furniture" onClick={this.getCategory} >FURNITURE</Link>
            <Link name="category" className="ml-auto text-white pr-5" style={{fontSize:22,textDecoration:"none"}} value="Clothing" onClick={this.getCategory}>CLOTHING</Link> 
          
          </Toolbar>
        </AppBar>
        <div style={{backgroundColor:"#DEE0FC"}}>
        {/* <Carousel id="carousel" animation="slide" navButtonsAlwaysVisible>
              <img src={require("../assets/img/clothing.png")} className="d-block offset-md-2 mt-4" width="1000" height="485" alt="Clothing Pic" />
              <img src={require("../assets/img/furnitures.png")} className="d-block offset-md-2 mt-4" width="1000" height="485" alt="Furniture Pic" />
              <img src={require("../assets/img/shoes.png")} className="d-block offset-md-2 mt-4" width="1000" height="485" alt="Shoes Pic" />
              <img src={require("../assets/img/mobiles.png")} className="d-block offset-md-2 mt-4" width="1000" height="485" alt="Mobiles Pic" />
        </Carousel> */}
        </div>
      </React.Fragment>
    );
     }
     else{
       return (
        <React.Fragment>
          {/* Slider */}
          <AppBar position="static">
            <Toolbar>
              
              <Link name="category" className="text-white pl-5" style={{fontSize:22,textDecoration:"none"}} value="Electronics" onClick={this.getCategory}>ELECTRONICS</Link>
              <Link name="category"  className="ml-auto text-white" style={{fontSize:22,textDecoration:"none"}} value="Shoes" onClick={this.getCategory} >SHOES</Link>
              <Link name="category" className="ml-auto text-white" style={{fontSize:22,textDecoration:"none"}} value="Furniture" onClick={this.getCategory} >FURNITURE</Link>
              <Link name="category" className="ml-auto text-white pr-5" style={{fontSize:22,textDecoration:"none"}} value="Clothing" onClick={this.getCategory}>CLOTHING</Link> 
            
            </Toolbar>
          </AppBar>
          <div style={{backgroundColor:"#DEE0FC"}}>
          {this.viewDetails()}
          </div>
        </React.Fragment>
      );
     }
  }
}

export default Home;