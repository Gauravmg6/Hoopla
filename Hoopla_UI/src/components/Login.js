import React, { Component } from "react";
import { Link } from 'react-router-dom';
import Avatar from '@material-ui/core/Avatar';
import Button from '@material-ui/core/Button';
import CssBaseline from '@material-ui/core/CssBaseline';
import FormControl from '@material-ui/core/FormControl';
import Input from '@material-ui/core/Input';
import InputLabel from '@material-ui/core/InputLabel';
import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import withStyles from '@material-ui/core/styles/withStyles';
import axios from "axios";

const usersBackendURL = "http://localhost:3000";

const styles = theme => ({
  main: {
    width: 'auto',
    display: 'block', // Fix IE 11 issue.
    marginLeft: theme.spacing(3),
    marginRight: theme.spacing(3),
    [theme.breakpoints.up(400 + theme.spacing(3) * 2)]: {
      width: 400,
      marginLeft: 'auto',
      marginRight: 'auto',
    },
  },
  paper: {
    marginTop: theme.spacing(5),
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    padding: `${theme.spacing(2)}px ${theme.spacing(3)}px ${theme.spacing(3)}px`,
  },
  avatar: {
    margin: theme.spacing(1),
    backgroundColor: theme.palette.secondary.main,
  },
  form: {
    width: '100%', // Fix IE 11 issue.
    marginTop: theme.spacing(1),
  },
  submit: {
    marginTop: theme.spacing(3),
  },
});

class LoginComponent extends Component {
  state = {
    form: {
      email: '',
      password: '',
    },
    formErrMsg: {
      email: '',
      password: '',
    },
    formValid: {
      email: false,
      password: false,
      buttonActive: false
    },
    successResponse: "",
    errorMessage: ""
  }


  handleInputChange = (e) => {
    const name = e.target.name;
    const value = e.target.value;
    this.setState({ form: { ...this.state.form, [name]: value } });
    this.validateField(name, value);
  }

  validateField = (fieldName, value) => {
    var message;
    var { formErrMsg } = this.state;
    var { formValid } = this.state;

    switch (fieldName) {
      case 'email':
        let emailRegex = new RegExp(/^[A-z][A-z0-9.]+@[a-z]+\.[a-z]{2,3}$/);
        value === "" ? message = "Please enter your email id" : emailRegex.test(value) ? message = "" : message = "Email id format is wrong"
        break;

      case "password":
        let passRegex = new RegExp(/^(?=.*[A-Z])(?=.*[!@#$&*%&])(?=.*[0-9])(?=.*[a-z]).{7,20}$/);
        value === "" ? message = "Please enter your password" : passRegex.test(value) ? message = "" : message = "Invalid password"
        break

      default:
        break;
    }
    //Form err message set
    formErrMsg[fieldName] = message;
    this.setState({ formErrMsg: formErrMsg });
    //Form Valid set
    message === "" ? formValid[fieldName] = true : formValid[fieldName] = false;
    formValid.buttonActive = formValid.email && formValid.password;
    this.setState({ formValid: formValid });
  }

  submitSignIn = (e) => {
    e.preventDefault();
    this.setState({successResponse:"",errorMessage:"",returnMessage:""})
    const userLoginData = { uEmail: this.state.form.email, uPass: this.state.form.password };
    axios.post(usersBackendURL + '/login', userLoginData)
      .then(response =>{this.setState({ successResponse: response.data})
      sessionStorage.setItem("login",response.data.uProfile.uName);
      sessionStorage.setItem("msg","You are logged in Go to Dashboard")
      window.location.reload(false)
      })
      .catch(error => {
        this.setState({ errorMessage: error.response ? error.response.data.message : error.message })
      })
  }

  render() {
    const { email, password } = this.state.form;
    const { formErrMsg } = this.state
    const { classes } = this.props;
    return (
      <React.Fragment>
        <br/>
        <h4 className="text-success text-center">{sessionStorage.getItem("msg")}</h4>
        <div className="col-md-4 offset-4">
          <main className={classes.main}>
            <CssBaseline />
            <Paper className={classes.paper}>
              <Avatar className={classes.avatar}>
                <LockOutlinedIcon />
              </Avatar>
              <Typography component="h1" variant="h5">
                Please login to continue
              </Typography>

              <form className={classes.form} onSubmit={this.submitSignIn}>
                <FormControl margin="normal" required fullWidth>
                  <InputLabel htmlFor="uemail">Email address</InputLabel>
                  <Input autoComplete="email" autoFocus
                    id="uEmail" name="email"
                    value={email} onChange={this.handleInputChange} />
                  <span className="text-danger">{formErrMsg.email}</span>
                </FormControl>

                <FormControl margin="normal" required fullWidth>
                  <InputLabel htmlFor="uPass">Password</InputLabel>
                  <Input name="password" type="password" value={password}
                    onChange={this.handleInputChange} id="uPass"
                    autoComplete="current-password" />
                  <span className="text-danger">{formErrMsg.password}</span><br />
                </FormControl>

                <Button
                  id="logButton"
                  type="submit"
                  fullWidth
                  variant="contained"
                  color="primary"
                  className={classes.submit}
                  disabled={!this.state.formValid.buttonActive}
                >Login
              </Button>

              </form><br />
              <Link to="/register" exact={"true"} onClick={this.handleClick}>New to Hoopla? Create an account</Link><br />
              {this.state.errorMessage ? (<div className={'text-danger'}>{this.state.errorMessage}</div>) :
                sessionStorage.getItem("msg") && <div className="text-success">{sessionStorage.getItem("msg")}</div>}
            </Paper>
          </main>
        </div>
      </React.Fragment>
    )
  }
}

export default withStyles(styles)(LoginComponent)