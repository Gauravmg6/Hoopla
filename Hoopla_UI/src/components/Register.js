import React, { Component } from "react";
import { Link } from 'react-router-dom';
import Avatar from '@material-ui/core/Avatar';
import Button from '@material-ui/core/Button';
import CssBaseline from '@material-ui/core/CssBaseline';
import FormControl from '@material-ui/core/FormControl';
import Input from '@material-ui/core/Input';
import InputLabel from '@material-ui/core/InputLabel';
import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import withStyles from '@material-ui/core/styles/withStyles';
import axios from "axios";
import Home from "./Home";

const usersBackendURL = "http://localhost:3000";

const styles = theme => ({
  main: {
    width: 'auto',
    display: 'block', // Fix IE 11 issue.
    marginLeft: theme.spacing(3),
    marginRight: theme.spacing(3),
    [theme.breakpoints.up(400 + theme.spacing(2) * 2)]: {
      width: 400,
      marginLeft: 'auto',
      marginRight: 'auto',
    },
  },
  paper: {
    marginTop: theme.spacing(1),
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    padding: `${theme.spacing(2)}px ${theme.spacing(3)}px ${theme.spacing(2)}px`,
  },
  avatar: {
    margin: theme.spacing(1),
    backgroundColor: theme.palette.secondary.main,
  },
  form: {
    width: '100%', // Fix IE 11 issue.
    marginTop: theme.spacing(1),
  },
  submit: {
    marginTop: theme.spacing(1),
  },
});

class RegisterComponent extends Component {
  state = {
    form: {
        userName:"",
        dob:"",
        mobileNumber:"",
        email: '',
        password: ''
    },
    formErrMsg: {
        userName:"",
        dob:"",
        mobileNumber:"",
        email: '',
        password: ''
    },
    formValid: {
        userName:false,
        dob:false,
        mobileNumber:false,
        email: false,
        password: false,
        buttonActive: false
    },
    successResponse: "",
    errorMessage: ""
  }


  handleInputChange = (e) => {
    const name = e.target.name;
    const value = e.target.value;
    this.setState({ form: { ...this.state.form, [name]: value } });
    this.validateField(name, value);
  }

  validateField = (fieldName, value) => {
    var message;
    var { formErrMsg } = this.state;
    var { formValid } = this.state;

    switch (fieldName) {
      case 'email':
        let emailRegex = new RegExp(/^[A-z][A-z0-9.]+@[a-z]+\.[a-z]{2,3}$/);
        value === "" ? message = "Please enter your email id" : emailRegex.test(value) ? message = "" : message = "Email id format is wrong"
        break;

      case "password":
        let passRegex = new RegExp(/^(?=.*[A-Z])(?=.*[!@#$&*%&])(?=.*[0-9])(?=.*[a-z]).{7,20}$/);
        value === "" ? message = "Please enter your password" : passRegex.test(value) ? message = "" : message = "Invalid password"
        break
    
      case "userName":
        if(value===""){
          message="Please enter your Name"
        }
        else if(!(value.match(/^[a-zA-z]+([\s][a-zA-Z]+)*$/))){
          message="Only alphabets. Should not start and end with space"
        }
        else message = ""
        break

      case "dob":
        if(value===""){
          message="Please enter your DOB"
        }
        else if(!(new Date(value)<new Date())){
          message="Date should be before Today(s) Date"
        }
        else message = ""
        break

      case "mobileNumber":
        if(value===""){
          message="Please Enter Your Mobile Number"
        }
        else if(!( String(value)[0]!="0")){
          message="Should not Start With 0"
        }
        else if(!(value.length==10)){
          message="Mobile Number should be 10 digits"
        }
        else message = ""
        break

      default:
        break;
    }
    //Form err message set
    formErrMsg[fieldName] = message;
    this.setState({ formErrMsg: formErrMsg });
    //Form Valid set
    message === "" ? formValid[fieldName] = true : formValid[fieldName] = false;
    formValid.buttonActive = formValid.email && formValid.password && formValid.userName && formValid.dob && formValid.mobileNumber;
    this.setState({ formValid: formValid });
  }

  submitSignIn = (e) => {
    e.preventDefault();
    this.setState({successResponse:"",errorMessage:""})
    axios.post(usersBackendURL + '/register',this.state.form)
      .then(response => this.setState({ successResponse: response.data }))
      .catch(error => {
        this.setState({ errorMessage: error.response ? error.response.data.message : error.message })
      })
  }

  render() {
    const { email, password,userName,dob,mobileNumber } = this.state.form;
    const { formErrMsg } = this.state
    const { classes } = this.props;
  if(this.state.successResponse===""){
    return (
      <React.Fragment>
        <div className="col-md-4 offset-4">
          <main className={classes.main}>
            <CssBaseline />
            <Paper className={classes.paper}>
              <Avatar className={classes.avatar}>
                <LockOutlinedIcon />
              </Avatar>
              <Typography component="h1" variant="h5">
                Create account
              </Typography>

              <form className={classes.form} onSubmit={this.submitSignIn}>
                <FormControl margin="normal" required fullWidth>
                  <InputLabel htmlFor="userName">Username</InputLabel>
                  <Input autoComplete="userName" autoFocus
                    id="userName" type="text" name="userName"
                    value={userName} onChange={this.handleInputChange} />
                  <span className="text-danger">{formErrMsg.userName}</span>
                </FormControl>

                <FormControl margin="normal" required fullWidth>
                  DOB<Input name="dob" type="date" value={dob}
                    onChange={this.handleInputChange} id="dob"
                    autoComplete="DOB" />
                  <span className="text-danger">{formErrMsg.dob}</span>
                </FormControl>

                <FormControl margin="normal" required fullWidth>
                  <InputLabel htmlFor="mobileNumber">Mobile Number</InputLabel>
                  <Input name="mobileNumber" type="number" value={mobileNumber}
                    onChange={this.handleInputChange} id="mobileNumber"
                    autoComplete="mobile-number" />
                  <span className="text-danger">{formErrMsg.mobileNumber}</span>
                </FormControl>

                <FormControl margin="normal" required fullWidth>
                  <InputLabel htmlFor="email">Email Address</InputLabel>
                  <Input name="email" type="email" value={email}
                    onChange={this.handleInputChange} id="uEmail"
                    autoComplete="email-address" />
                  <span className="text-danger">{formErrMsg.email}</span>
                </FormControl>

                <FormControl margin="normal" required fullWidth>
                  <InputLabel htmlFor="password">Password</InputLabel>
                  <Input name="password" type="password" value={password}
                    onChange={this.handleInputChange} id="uPass"
                    autoComplete="password" />
                  <span className="text-danger">{formErrMsg.password}</span>
                </FormControl>

                <Button
                  id="regButton"
                  type="submit"
                  fullWidth
                  variant="contained"
                  color="primary"
                  className={classes.submit}
                  disabled={!this.state.formValid.buttonActive}
                >Register
              </Button>
              <div className="text-danger">{this.state.errorMessage}</div>
              </form><br />
              <Link to="/login" exact={"true"} onClick={this.handleClick}>Already Registeres? Login Here</Link><br />
            </Paper>
          </main>
        </div>
      </React.Fragment>
    )}
    else return <Home></Home>
  }
}

export default withStyles(styles)(RegisterComponent)