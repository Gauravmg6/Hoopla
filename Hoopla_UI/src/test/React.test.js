import React from 'react';
import {configure} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({adapter: new Adapter()});
import {shallow,mount} from "enzyme";
import {create} from "react-test-renderer";


import Home from "../components/Home";
import Login from "../components/Login";
import Register from "../components/Register";
import Navbar from "../components/NavBar";
import Cart from "../components/cart";

import Carousel from 'react-material-ui-carousel';
import { Link } from 'react-router-dom';


describe("Snapshot",()=>{
    test("Register snapshot",()=>{
        const component=create(<Cart />);
        let tree=component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe("Home Component",()=>{
    test("existence",()=>{
        const wrapper=shallow(<Home />)
        expect(wrapper.exists()).toBe(true);
    });
    test("FormControl Length check",()=>{
        const wrapper=mount(<Home />)
        expect(wrapper.find(Carousel).length).toEqual(1);
    });
    test("FormControl Length check",()=>{
        const wrapper=mount(<Home />)
        expect(wrapper.find(Link).length).toEqual(4);
    });
});

describe("Login Component",()=>{
    test("existence",()=>{
        const wrapper=shallow(<Login />)
        expect(wrapper.exists()).toBe(true);
    });
    test("props Check",()=>{
        const wrapper=shallow(<Login />)
        expect(wrapper.props()).toBeTruthy();
    });
});

describe("register component",()=>{
    test("existence",()=>{
        const wrapper=shallow(<Register />)
        expect(wrapper.exists()).toBe(true);
    });
    test("props check",()=>{
        const wrapper=shallow(<Register />)
        expect(wrapper.props()).toBeTruthy();
    });
});

describe("Navbar component",()=>{
    test("existence",()=>{
        const wrapper=shallow(<Navbar />)
        expect(wrapper.exists()).toBe(true);
    });
});

