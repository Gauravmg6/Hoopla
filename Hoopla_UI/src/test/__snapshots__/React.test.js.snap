// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Register snapshot 1`] = `
<div
  className="mt-5 offset-4"
  id="image"
>
  <img
    src="cartEmpty.png"
    style={
      Object {
        "marginTop": "120px",
      }
    }
  />
</div>
`;
